File: PYTHON/BAEKJOON/2750_수_정렬하기(버블,선택,삽입)/2750_BubbleSort.py
def bubble_sort(arr, N):
    for i in range(N - 1, 0, -1):
        for j in range(0, i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

    return arr


N = int(input())
arr = [int(input()) for _ in range(N)]
for v in bubble_sort(arr, N):
    print(v)

File: BAEKJOON/2750_수_정렬하기(버블,선택,삽입)/2750_BubbleSort.js
const fs = require('fs');
const fileStream = fs.createReadStream('input.txt');
const readline = require('readline');

const rl = readline.createInterface({
  input: fileStream,
  output: process.stdout,
})

let N;
let input = []
let lineIdx = 0

function bubbleSort(arr, N) {
  // 마지막이 정렬되는건 선택 정렬이랑 똑같다.
  for (let i = N - 1; i > 0; i--) {
    // 처음부터 i까지 서로 비교해가면서 바꿔간다.
    for (let j = 0; j < i; j++) {
      // swap
      let temp = arr[j];
      if (arr[j] > arr[j + 1]) {
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }

  return arr;
}

rl.on("line", (line) => {
  if (lineIdx === 0) {
    N = Number(line)
    lineIdx++;
  } else {
    input.push(Number(line))
  }
})
  .on("close", () => {
    // 로직
    bubbleSort(input, N).forEach(v => console.log(v));
    process.exit()
  })